src/CommunityToolkit/Xamarin.CommunityToolkit/Views/AutoFitLabel/Android/AutoFitLabelRenderer.android.cs

```diff
@@ -0,0 +1,792 @@
+
+			var widthMeasureSpec = MeasureSpecFactory.MakeMeasureSpec(realWidth, MeasureSpecMode.Exactly);
+			var heightMeasureSpec = MeasureSpecFactory.MakeMeasureSpec(realHeight, MeasureSpecMode.Exactly);
+
+			Control.Measure(widthMeasureSpec, heightMeasureSpec);
+		}
+
+		public void RecalculateSpanPositions(Label element, SpannableString spannableString, SizeRequest finalSize)
+		{
+			if (element?.FormattedText?.Spans == null || element.FormattedText.Spans.Count == 0)
+				return;
+
+			var labelWidth = finalSize.Request.Width;
+			if (labelWidth <= 0 || finalSize.Request.Height <= 0)
+				return;
+
+			if (spannableString == null || spannableString.IsDisposed())
+				return;
+
+			var layout = Layout;
+			if (layout == null)
+				return;
+
+			var next = 0;
+			var count = 0;
+			var totalLineHeights = new List<int>();
+
+			for (var i = 0; i < spannableString.Length(); i = next)
+			{
+				var type = Java.Lang.Class.FromType(typeof(Java.Lang.Object));
+
+				var span = element.FormattedText.Spans[count];
+
+				count++;
+
+				if (string.IsNullOrEmpty(span.Text))
+					continue;
+
+				// find the next span
+				next = spannableString.NextSpanTransition(i, spannableString.Length(), type);
+
+				// get all spans in the range - Android can have overlapping spans
+				var spans = spannableString.GetSpans(i, next, type);
+
+				if (spans is null)
+					continue;
+
+				var startSpan = spans[0];
+				var endSpan = spans[^1];
+
+				var startSpanOffset = spannableString.GetSpanStart(startSpan);
+				var endSpanOffset = spannableString.GetSpanEnd(endSpan);
+
+				var thisLine = layout.GetLineForOffset(endSpanOffset);
+				var lineStart = layout.GetLineStart(thisLine);
+				var lineEnd = layout.GetLineEnd(thisLine);
+
+				// If this is true, endSpanOffset has the value for another line that belong to the next span and not it self.
+				// So it should be rearranged to value not pass the lineEnd.
+				if (endSpanOffset > (lineEnd - lineStart))
+					endSpanOffset = lineEnd;
+
+				var startX = layout.GetPrimaryHorizontal(startSpanOffset);
+				var endX = layout.GetPrimaryHorizontal(endSpanOffset);
+
+				var startLine = layout.GetLineForOffset(startSpanOffset);
+				var endLine = layout.GetLineForOffset(endSpanOffset);
+
+				var lineHeights = new double[endLine - startLine + 1];
+
+				// calculate all the different line heights
+				for (var lineCount = startLine; lineCount <= endLine; lineCount++)
+				{
+					var lineHeight = layout.GetLineBottom(lineCount) - layout.GetLineTop(lineCount);
+					lineHeights[lineCount - startLine] = lineHeight;
+
+					if (totalLineHeights.Count <= lineCount)
+						totalLineHeights.Add(lineHeight);
+				}
+
+				var yaxis = 0.0;
+				for (var line = startLine; line > 0; line--)
+					yaxis += totalLineHeights[line];
+
+				((ISpatialElement)span).Region = Forms.Region.FromLines(lineHeights, labelWidth, startX, endX, yaxis).Inflate(10);
+			}
+		}
+
+		void SetLineBreakMode(Label label)
```

Comment:
Can we use the Forms methods? For this and other methods below 
https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.Android/Extensions/TextViewExtensions.cs